src/state/selectors/index.js

```diff
@@ -96,6 +98,49 @@ export function getSelectedCanvas(state, windowId) {
       .getCanvasByIndex(canvasIndex);
 }
 
+/**
+* Return the current canvas' (selected in a window) annotations
+* @param {object} state
+* @param {String} windowId
+* @return {Array}
+*/
+export function getSelectedCanvasAnnotations(state, canvasId) {
+  const annotations = state.annotations && state.annotations[canvasId];
+  if (!annotations) return [];
+
+  return filter(
+    annotations,
+    annotation => annotation
+                  && annotation.json
+                  && annotation.json.resources
+                  && annotation.json.resources.length,
+  );
+}
+
+/**
+* Return an array of annotation resources filtered by the given motivation
+* @param {Array} annotations
+* @param {Array} motivations
+* @return {Array}
+*/
+export function getAnnotationResourcesByMotivation(annotations, motivations) {
+  const resources = flatten(annotations.map(annotation => annotation.json.resources));
+
+  return filter(resources, resource => flatten(new Array(resource.motivation)).some(
+    motivation => motivations.includes(motivation),
+  ));
+}
+
+/**
+ * @param {Array} annotations
+ * @return {Array} [{ id: 'abc123', content: 'Annotation Content' }, ...]
+ */
+export function getIdAndContentOfAnnotations(annotations) {
+  return annotations.map((annotation, i) => ({
+    id: (annotation['@id'] || `annotation-${i}`),
+    content: flatten(new Array(annotation.resource)).map(r => r.chars).join(' '),
+  }));
+}
```

Comment:
I wasn't sure if these nested functions was the way to go.  It almost feels like I want to extract a class similar to what ManifestoCanvas does, but kept it simple for now.  Happy to work on that, split that out into a separate issue, or forget about it.

Reply:
Also, this is pretty much tied to the V2 (Open Annotation) format.  Should I be doing anything to either `try`/`catch` somewhere or explicitly filter for only Open Annotations (based on `@context`).?

Reply:
not clear to me why `canvasId` is being used as the state reference for the annotations.  Passing it as a parameter to the fetch action is somewhat questionable, since it is not used by the fetch action, and is essentially concatenated into the API response by the reducer.  An annotation is self-contained and already contains the target to which it references.  Filtering stored annotations by target would be a more extensible approach, since it is theoretically possible for an annotation to refer to anything,

Reply:
@christopher-johnson please create a spin off issue for this and link any fixture data that may be relevant or helpful for an implementer.